### PR TITLE
add square pointer

### PR DESCRIPTION
**Pull request by MaraScott**
provide a different shape for the mask.
it is useful when you want to make a square mask specificaly

### CODE REVIEW

**Comment by MaraScott** [comment]
original pull request https://github.com/comfyanonymous/ComfyUI/pull/3561#issuecomment-2351970948

333603882-fc5bd3fd-9835-488a-926e-7de7d695bcf5

File: src/extensions/core/maskeditor.ts [hunk 1]
```diff
@@ -120,6 +120,8 @@ class MaskEditorDialog extends ComfyDialog {
   mousedown_pan_x: number
   mousedown_pan_y: number
   last_pressure: number
+  pointer_type: string
```
Comment (huchenlei) [inline src/extensions/core/maskeditor.ts]: Define a enum for `pointer_type`.

```js
enum PointerType {
  Arc = 'arc'
  Rect = 'rect'
}
```

File: src/extensions/core/maskeditor.ts [hunk 2]
```diff
@@ -818,7 +887,16 @@ class MaskEditorDialog extends ComfyDialog {
           self.maskCtx.beginPath()
           self.maskCtx.fillStyle = this.getMaskFillStyle()
           self.maskCtx.globalCompositeOperation = 'source-over'
-          self.maskCtx.arc(x, y, brush_size, 0, Math.PI * 2, false)
+          if (self.pointer_type === 'rect') {
```
Comment (huchenlei) [inline src/extensions/core/maskeditor.ts]: Extract this logic into a function, as it is repeated over 3 times.

**Review by huchenlei — CHANGES_REQUESTED** [review]

File: src/extensions/core/maskeditor.ts (hunk 1 above)
Comment (MaraScott) [reply src/extensions/core/maskeditor.ts]: this is corrected

File: src/extensions/core/maskeditor.ts (hunk 2 above)
Comment (MaraScott) [reply src/extensions/core/maskeditor.ts]: this is corrected

**Review by huchenlei — APPROVED** [review]
LGTM

**Comment by MaraScott** [comment]
Super cool thanks

**Pull request merged**